### PR TITLE
feat: use commander as arg-parser

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
- Closes: #748 

- Blocked by: #861

### CODE REVIEW

File: packages/cli/src/commands/app.ts [hunk 1]
```diff
@@ -1,210 +1,108 @@
...
+      )
+      .option(
+        `-i, --ipfs [<${intlMsg.commands_codegen_options_i_node()}>] `,
+        `${intlMsg.commands_codegen_options_i()}`
+      )
+      .option(
+        `-e, --ens [<${intlMsg.commands_codegen_options_e_address()}>]`,
+        `${intlMsg.commands_codegen_options_e()}`
+      )
+      .action(async (options) => {
```
Comment (Niraj-Kamdar) [inline packages/cli/src/commands/app.ts:57]: This manual parsing isn't needed but it seems like there is some issue with `commander` parser handler so until it gets fixed, we will be using manual parsing.

File: packages/cli/src/__tests__/e2e/app.spec.ts [hunk 2]
```diff
@@ -111,6 +105,9 @@ ${HELP}`);
           },
         );
 
+        console.log(output);
```
Comment (nerfZael) [inline packages/cli/src/__tests__/e2e/app.spec.ts]: I think you missed these :D

File: packages/cli/src/__tests__/e2e/deploy.spec.ts [hunk 3]
```diff
@@ -15,13 +15,15 @@ import yaml from "js-yaml";
...
-w3 deploy [options]
+const HELP = `Usage: w3 deploy|d [options]
+
+Deploys/Publishes a Web3API
 
 Options:
-  -h, --help                         Show usage information
-  -m, --manifest-file <path>         Path to the Web3API Deploy manifest file (default: web3api.yaml | web3api.yml)
-  -v, --verbose                      Verbose output (default: false)
+  -m, --manifest-file <path>  Path to the Web3API Plugin manifest file
```
Comment (nerfZael) [inline packages/cli/src/__tests__/e2e/deploy.spec.ts]: Shouldn't this be "Path to the Web3API wrapper manifest" ?

File: packages/cli/src/commands/app.ts [hunk 4]
```diff
@@ -1,210 +1,108 @@
...
-  fixParameters,
-  resolvePathIfExists,
-  defaultAppManifest,
   getSimpleClient,
   getTestEnvProviders,
 } from "../lib";
+import {
+  parseAppManifestFileOption,
+  parseAppCodegenDirOption,
+  // defaultAppCodegenDirOption,
```
Comment (nerfZael) [inline packages/cli/src/commands/app.ts]: What's up with the comments?

File: packages/cli/src/commands/create.ts [hunk 5]
```diff
@@ -1,182 +1,143 @@
...
+      filesystem.remove(projectDir);
+    } else {
+      process.exit(8);
+    }
   }
 
-  return true;
+  generateProjectTemplate(command, lang, projectDir, filesystem)
+    .then(() => {
+      console.log();
```
Comment (nerfZael) [inline packages/cli/src/commands/create.ts:126]: Leftover console log?

**Review by nerfZael — CHANGES_REQUESTED** [review]

**Comment by nerfZael** [comment]
Do we still need gluegun, or could we completely remove it?

File: packages/cli/src/__tests__/e2e/deploy.spec.ts (hunk 3 above)
Comment (Niraj-Kamdar) [reply packages/cli/src/__tests__/e2e/deploy.spec.ts]: Nice catch!

File: packages/cli/src/commands/app.ts (hunk 4 above)
Comment (Niraj-Kamdar) [reply packages/cli/src/commands/app.ts]: That's for fixing it back when commander default arg-parser start working.

Comment (Niraj-Kamdar) [reply packages/cli/src/commands/app.ts]: I can remove it though for now

File: packages/cli/src/commands/create.ts (hunk 5 above)
Comment (Niraj-Kamdar) [reply packages/cli/src/commands/create.ts:126]: Nope that's for output formatting

**Comment by Niraj-Kamdar** [comment]
> Do we still need gluegun, or could we completely remove it?

Yep but it would be better to do it in other PR.

File: packages/cli/src/commands/app.ts (hunk 4 above)
Comment (nerfZael) [reply packages/cli/src/commands/app.ts]: Or just add "//TODO: ..."
Or better yet: "//TODO: <issue number>"

**Review by dOrgJelli — APPROVED** [review]

**Pull request merged**